### PR TITLE
OCPCLOUD-2584: Add manifest generation

### PR DESCRIPTION
**Pull request by RadekManak**
This PR adds a manifest generator tool that creates manifest files with the infrastructure provider component.

It also adds a temporary carry patch that disables inclusion of ASO component. 
We currently cannot deploy ASO because it causes a conflict with cluster-capi-operator. Both controllers want to modify CRDs. ASO to the upstream variant and CCAPI to the customized versions with OpenShift labels.  We might need to add a patch to ASO to not manage CRDs or not deploy ASO CRDs from cluster-capi-operator and let ASO do it.

### CODE REVIEW

File: openshift/Dockerfile.openshift [hunk 1]
```diff
@@ -4,15 +4,16 @@ WORKDIR /build
 COPY . .
 RUN GO111MODULE=on CGO_ENABLED=0 GOOS=${GOOS} GOPROXY=${GOPROXY} go build \
   -ldflags="-w -s -X 'main.version=${VERSION}'" \
-  -o=cluster-api-provider-azure-controller-manager \
+  -o=manager \
```
Comment (damdo) [inline openshift/Dockerfile.openshift:7]: If you change this (which I'm happy with) we should also change the one over in the capiinstaller here: https://github.com/openshift/cluster-capi-operator/blob/3f3e845f5daf0d1fc70a188117152660e5de18c4/pkg/controllers/capiinstaller/component_customizer.go#L41

**Review by damdo — COMMENTED** [review]
Mostly LGTM.
Are we expecting to have openshift/manifests/0000_30_cluster-api_04_cm.infrastructure-azure.yaml compressed? Or not because of ASO not being present?

File: openshift/Dockerfile.openshift (hunk 1 above)
Comment (RadekManak) [reply openshift/Dockerfile.openshift:7]: Good catch. Changed in the cluster-capi-operator PR.

**Comment by RadekManak** [comment]
Tested cluster with both PRs and ran E2E on it. The machine was created successfully.

This PR is safe to merge, as the controller is not deployed until the other PR merges.

**Pull request merged**